packages/lambda-powertools-middleware-correlation-ids/index.d.ts

```diff
@@ -0,0 +1,61 @@
+import middy from "@middy/core";
+import {
+  Context,
+  SQSRecord,
+  Handler,
+  DynamoDBRecord,
+  DynamoDBStreamEvent,
+  Callback,
+  FirehoseTransformationEvent,
+  FirehoseTransformationResult,
+  KinesisStreamEvent,
+} from "aws-lambda";
+import Log from "@dazn/lambda-powertools-logger";
+import CorrelationIds from "@dazn/lambda-powertools-correlation-ids";
+
+export default function <T, R, C extends Context = Context>(params: {
+  sampleDebugLogRate: number;
+}): middy.MiddlewareObject<T, R, C>;
+
+export type ExtractedCorrelationIdAndLogger = {
+  logger: Log;
```

Comment:
@simontabor with the comment below, about not being coupled with the dazn logger,  does this mean you would prefer us to use a Generic here?